src/sentry/integrations/slack/utils/notifications.py

```diff
@@ -63,12 +70,78 @@ def send_incident_alert_notification(
+        )
+    except Exception:
+        # if there's an error trying to grab a parent notification, don't let that error block this flow
+        pass
+
+    new_notification_message_object = NewMetricAlertNotificationMessage(
+        incident_id=incident.id,
+        trigger_action_id=action.id,
+    )
+
+    # If a parent notification exists for this rule and action, then we can reply in a thread
+    if parent_notification_message is not None:
+        # Make sure we track that this reply will be in relation to the parent row
+        new_notification_message_object.parent_notification_message_id = (
+            parent_notification_message.id
+        )
+
+        # Only reply and use threads if the organization is enabled in the FF
+        if features.has("organizations:slack-thread", incident.organization):
+            # To reply to a thread, use the specific key in the payload as referenced by the docs
+            # https://api.slack.com/methods/chat.postMessage#arg_thread_ts
+            payload["thread_ts"] = parent_notification_message.message_identifier
+
+    success = False
+    try:
+        response = client.post("/chat.postMessage", data=payload, timeout=5)
+        # response should include a "ts" key that represents the unique identifier for the message
+        # referenced at https://api.slack.com/methods/chat.postMessage#examples
+        success = True
+    except ApiError as e:
+        # Record the error code and details from the exception
+        new_notification_message_object.error_code = e.code
+        new_notification_message_object.error_details = {
+            "url": e.url,
+            "host": e.host,
+            "path": e.path,
+            "data": e.json if e.json else e.text,
+        }
         logger.info("rule.fail.slack_post", exc_info=True)
-    return False
+    else:
+        # Slack will always send back a ts identifier https://api.slack.com/methods/chat.postMessage#examples
+        # on a successful message
+        ts = None
+        # This is a workaround for typing, and the dynamic nature of the return value
+        if isinstance(response, BaseApiResponse):
+            ts = response.json.get("ts")
+        elif isinstance(response, MappingApiResponse):
+            ts = response.get("ts")
+        else:
+            logger.info(
+                "failed to get ts from slack response",
+                extra={
+                    "response_type": type(response).__name__,
+                },
+            )
+        new_notification_message_object.message_identifier = str(ts) if ts is not None else None
+
+    # Save the notification message we just sent with the response id or error we received
+    try:
+        repository.create_notification_message(data=new_notification_message_object)
+    except Exception:
+        # If we had an unexpected error with saving a record to our datastore,
+        # do not let the error bubble up, nor block this flow from finishing
```

Comment:
Anywhere were we're fetching generic Exception, it's good to have logging, otherwise there will be silent failures with no way to debug.